pom.xml

```diff
@@ -238,7 +218,7 @@
             <version>2.1.0</version>
         </dependency>
 
-        <!-- Used for loading/fetching/writing GTFS entities -->
+        <!-- Used for loading/fetching/writing GTFS entities (also provides access to commons-io and AWS S3 SDK). -->
```

Comment:
I don't think that this is good practice.  What happens if this dependency decides to no longer depend on commons-io?  Is there a better way to go about reducing the jar size?

Reply:
If we updated that dependency and it no longer depended transitively on commons-io then we'd just need to add commons-io to our list of dependencies, which doesn't seem like a problem. The approach of always including our own dependency on commons-io (rather than relying on the transitive one) sometimes also seems to cause problems if the versions don't match.

If the version numbers match I also don't think this would reduce the JAR size. I don't expect it to include two shaded copies of the same dependency just because they're at different levels in the dependency hierarchy.

It's worth a review of exactly what Maven will do in the case that we depend transitively on two different versions of a library.

I don't know of any specific best practice in this department, we'd have to read other people's (probably highly conflicting) opinions on the subject or just do whatever works for us. I don't really mind either way as long as there are comments about what we're relying on transitively that will save us wasted time in the future when things go haywire.